### DISCUSSION TITLE
Trouble connecting GitHub with AWS CodeDeploy<title>

### DISCUSSION THREAD

**Discussion by senthil-ark** [community/community#45982] [Enterprise]
### Select Topic Area

Question

### Body

I'm trying to connect a Github Enterprise private repo with AWS CodeDeploy.  I am able to get it working with OAuth tokens, if "Third-party application access policy" is disabled in Github.  But when it is enabled, then AWS CodeDeploy cannot access the repository.  How can we add AWS CodeDeploy as an approved Third-party application?

These are the current apps which are approved in Github.  How to add AWS CodeDeploy here?
![Third-party app access policy](https://user-images.githubusercontent.com/108492557/215956104-0b341f27-9352-4b66-a8d2-52872873a6fc.jpg)

**Comment by jge162** [comment]
> ### Select Topic Area
> Question
> 
> ### Body
> I'm trying to connect a Github Enterprise private repo with AWS CodeDeploy. I am able to get it working with OAuth tokens, if "Third-party application access policy" is disabled in Github. But when it is enabled, then AWS CodeDeploy cannot access the repository. How can we add AWS CodeDeploy as an approved Third-party application?
> 
> These are the current apps which are approved in Github. How to add AWS CodeDeploy here? ![Third-party app access policy](https://user-images.githubusercontent.com/108492557/215956104-0b341f27-9352-4b66-a8d2-52872873a6fc.jpg)

seems strange

**Reply by jge162** [reply]
when you go to settings, developer and then oauth apps, do you see AWS codeply as an option?
<img width="1158" alt="Screenshot 2023-01-31 at 10 51 52 PM" src="https://user-images.githubusercontent.com/31228460/215972269-184b84f8-7ead-414d-bb4c-4739a2546968.png">

**Reply by senthil-ark** [reply]
Thanks jge162

**Comment by senthil-ark** [comment]
I setup "AWS" myself, but it's very likely I didn't set it up correctly

![Github OAuth Apps](https://user-images.githubusercontent.com/108492557/216139894-ec3c182c-1184-49d1-94cc-e3432c4b6e7b.jpg)

**Comment by senthil-ark** [comment]
After this screenshot, I created a new "client secret":
![Github AWS settings](https://user-images.githubusercontent.com/108492557/216141629-f85beb9a-ff41-425e-bada-140e78686f40.jpg)

So, I need to copy this secret into AWS somewhere?  Are the reset of these settings looking proper?

**Reply by jge162** [reply]
Settings look correct.

**Reply by senthil-ark** [reply]
What do we do with this secret (generated in Github)?

**Comment by senthil-ark** [comment]
So, I've set this up now. but still not able to sync from github to aws codedeploy

![AWS Github connection](https://user-images.githubusercontent.com/108492557/216159292-3da4d47c-cbdc-4714-ab3e-16efb5e401a9.jpg)
